### PR TITLE
Implement existence checks by calling DC API (needs to be enabled by -e)

### PR DESCRIPTION
**Pull request by pradh**
This checks that references to schema-like things exist in the KG or the local graph (i.e., other provided MCFs).

Notably, for performance reasons, for now it does not validate references to place entities (`observationAbout` value).

Existence check results are cached locally to avoid future calls.  Batching would be nice, but that would be a more significant refactor to the call-sites (e.g., we need to save the per-reference context required for logging), so we can consider in future.

Testing: there's a unit test for ExistenceChecker and another to confirm that individual nodes are existence-checked in McfChecker. Plan to add an e2e once the LintTest is generalized.

TODO: get some real testing and enable `-e` by default...

### CODE REVIEW

File: util/src/main/java/org/datacommons/util/ExistenceChecker.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+    var dataJson = callDc(sub, pred);
+    logCtx.incrementCounterBy("Existence_NumDcCalls", 1);
+    if (dataJson == null) {
+      // If the DCID is malformed Mixer can return failure.
+      return false;
+    }
+    if (dataJson.entrySet().size() != 1) {
+      throw new IOException(
+          "Invalid payload content from Staging DC API endpoint for: '"
```
Comment (beets) [inline util/src/main/java/org/datacommons/util/ExistenceChecker.java:102]:
```suggestion
          "Invalid results payload from Staging DC API endpoint for: '"
```

**Review by beets — APPROVED** [review]
thanks, sorry for the late review!

**Review by chejennifer — APPROVED** [review]
Sorry for the late review, looks good!

File: util/src/main/java/org/datacommons/util/ExistenceChecker.java (hunk 1 above)
Comment (pradh) [reply util/src/main/java/org/datacommons/util/ExistenceChecker.java:102]: Done

**Pull request merged**